Prog/Crimes_forweb.sas

```diff
@@ -0,0 +1,86 @@
+/**************************************************************************
+ Program:  Crime_forweb
+ Library:  Police
+ Project:  NeighborhoodInfo DC
+ Author:   Rob Pitingolo
+ Created:  11/07/2017
+ Version:  SAS 9.4
+ Environment:  Windows
+ Modifications: 
+
+**************************************************************************/
+
+%include "L:\SAS\Inc\StdLocal.sas"; 
+
+** Define libraries **;
+%DCData_lib( Police )
+%DCData_lib( Web )
+
+
+/***** Update the let statements for the data you want to create CSV files for *****/
+
+%let library = police; /* Library of the summary data to be transposed */
+%let outfolder = crime; /* Name of folder where output CSV will be saved */
+%let sumdata = crimes_sum; /* Summary dataset name (without geo suffix) */
+%let start = 2000; /* Start year */
+%let end = 2016; /* End year */
+%let keepvars = crimes_pt1 Crimes_pt1_property Crimes_pt1_violent Crime_rate_pop; /* Summary variables to keep and transpose */
+
+
+/***** Update the web_varcreate marcro if you need to create final indicators for the website after transposing *****/
+
+%macro web_varcreate;
+
+Crimes_pt1_per1000 = crimes_pt1 / Crime_rate_pop*1000;
+Crimes_pt1_property_per1000 = Crimes_pt1_property / Crime_rate_pop*1000;
+Crimes_pt1_violent_per1000 = Crimes_pt1_violent / Crime_rate_pop*1000;
+
+%mend web_varcreate;
+
+
+
+/**************** DO NOT UPDATE BELOW THIS LINE ****************/
+
+%macro csv_create(geo);
+			 
+%web_transpose(&library., &outfolder., &sumdata., &geo., &start., &end., &keepvars. );
+
+/* Load transposed data, create indicators for profiles */
+data &sumdata._&geo._long_allyr;
+	set &sumdata._&geo._long;
+	%web_varcreate;
+run;
+
+/* Output all-years CSV */
+ods csv file ="&_dcdata_default_path.\web\output\&outfolder.\&geo.\&outfolder._&geo._AllYears.csv";
```

Comment:
I think this is fine but I have a question. When running  X statement with the MKDIR command, what happens if the folders already exist? The all-years file was for testing, so I can remove it now that the decision has been made.

Reply:
@rpitingolo If the folders already exist, MKDIR should not do anything to disturb them. I've used it in other programs and that's the way it works. 

Note that you will want to include an `options noxwait;` statement in the program so that it won't stop everytime it opens a command prompt.

Reply:
Yes. Definitely. I am going to open a bugfix issue in the web repo to edit the macro.